src/core_types.jl

```diff
@@ -1,14 +1,20 @@
 """
-    abstract Rotation{T} <: StaticMatrix{T}
+    abstract Rotation{N,T} <: StaticMatrix{T}
 
-An abstract type representing 3D rotations. More abstractly, they represent
-3×3 unitary (orthogonal) matrices.
+An abstract type representing `N`-dimensional rotations. More abstractly, they represent
+unitary (orthogonal) `N`×`N` matrices.
 """
-abstract Rotation{T} <: StaticMatrix{T}
+abstract Rotation{N,T} <: StaticMatrix{T}
 
-Base.size{R <: Rotation}(::Union{R,Type{R}}) = (3,3)
+typealias Rotation2{T} Rotation{2,T}
+typealias Rotation3{T} Rotation{3,T}
+
+Base.size{N}(::Type{Rotation{N}}) = (N,N)
```

Comment:
I see that in 99.9% of cases the `@pure` below will be used but I think it is *technically* possible that adding `@pure` here could improve type inference. 

(or that's just me being paranoid. Since this package, and StaticArrays, is for speed, using unstable types like `Vector{Rotation{3}}` probably wouldn't occur very often)